include/RAJA/pattern/launch/launch_core.hpp

```diff
@@ -291,28 +326,53 @@ launch(RAJA::resources::Resource res, LaunchParams const &params, const char *ke
 {
 
   ExecPlace place;
-  if(res.get_platform() == camp::resources::v1::Platform::host) {
+  if(res.get_platform() == RAJA::Platform::host) {
     place = RAJA::ExecPlace::HOST;
   }else{
     place = RAJA::ExecPlace::DEVICE;
   }
 
+  //
+  //Configure plugins
+  //
+#ifdef RAJA_DEVICE_ACTIVE
+  util::PluginContext context{place == ExecPlace::HOST ?
+      util::make_context<typename POLICY_LIST::host_policy_t>()
+      : util::make_context<typename POLICY_LIST::device_policy_t>()};
+#else
+  util::PluginContext context{util::make_context<typename POLICY_LIST::host_policy_t>()};
+#endif
+
+  util::callPreCapturePlugins(context);
+
+  using RAJA::util::trigger_updates_before;
+  auto p_body = trigger_updates_before(body);
+
+  util::callPostCapturePlugins(context);
+
+  util::callPreLaunchPlugins(context);
+
   switch (place) {
     case ExecPlace::HOST: {
       using launch_t = LaunchExecute<typename POLICY_LIST::host_policy_t>;
-      return launch_t::exec(res, params, kernel_name, body); break;
+      return launch_t::exec(res, params, kernel_name, p_body); break;
     }
 #ifdef RAJA_DEVICE_ACTIVE
     case ExecPlace::DEVICE: {
       using launch_t = LaunchExecute<typename POLICY_LIST::device_policy_t>;
-      return launch_t::exec(res, params, kernel_name, body); break;
+      return launch_t::exec(res, params, kernel_name, p_body); break;
     }
 #endif
     default: {
       RAJA_ABORT_OR_THROW("Unknown launch place or device is not enabled");
     }
   }
-  //Should not get here;
+
+  util::callPostLaunchPlugins(context);
```

Comment:
This is in the wrong place, it will never gets called. @artv3

Reply:
ah! you are right, I can can fix it within the next couple of hours (after the first meeting of the day).

Reply:
fixing in #1394